src/odemis/gui/comp/miccanvas.py

```diff
@@ -1165,6 +1166,110 @@ def update_drawing(self):
 
         if self.view:
             self.update_thumbnail()
+            
+class NavigableBarPlotCanvas(BarPlotCanvas):
+    
+    def __init__(self, *args, **kwargs):
+        super(NavigableBarPlotCanvas, self).__init__(*args, **kwargs)
+        self.abilities |= {CAN_DRAG}
+        
+        # When resizing, margin to put around the current size
+        # TODO: Maybe make the margin related to the canvas size?
+        self.default_margin = 512
+        self.margins = (self.default_margin, self.default_margin)
+
+        # the position the view is asking to the next buffer recomputation
+        # in buffer-coordinates: = 1px at scale = 1
+        self._data_buffer = None
+        self.window_x = None
+        self.window_y = None
+        self.display_xrange = None
+        self.display_yrange = None
+        self.data_xrange = None
+        self.data_yrange = None
+
+    def set_1d_data(self, xs, ys, unit_x=None, unit_y=None, range_x=None, range_y=None):
```

Comment:
Is calling `set_data()` still possible? (I guess not really). If not, please mention it in the docstring.